### PR TITLE
Bump containerd client

### PR DESCRIPTION
**Pull request by cpuguy83**
- Bumps containerd client
- Uses containerd client's new `Reconnect()` API to reconnect cached clients instead of replacing them (leaving stale clients in other cached objects).

### CODE REVIEW

File: libcontainerd/client_daemon.go [hunk 1]
```diff
@@ -114,6 +114,13 @@ type client struct {
 	containers map[string]*container
 }
 
+func (c *client) reconnect() error {
```
Comment (cpuguy83) [inline libcontainerd/client_daemon.go]: We probably need to catch connection failures in normal requests and trigger a reconnect, but we aren't handling these cases at all (or ever AFAIK).

**Comment by thaJeztah** [comment]
Will this change be part of the upcoming containerd 1.1.0 release? Wondering if we can get back to released versions of the dependency (or at least something on a release branch)

**Comment by cpuguy83** [comment]
I think so, a release branch has not be cut yet.

**Comment by cpuguy83** [comment]
The new client API is not really "new" anymore since it's been a couple of weeks, btw.

**Comment by thaJeztah** [comment]
Alright, so probably safe to assume it’ll be in 1.1.0, and once that’s released, we could switch to a tagged version, thanks!

**Comment by crosbymichael** [comment]
LGTM

**Review by mlaventure — APPROVED** [review]
LGTM

File: libcontainerd/client_daemon.go (hunk 1 above)
Comment (mlaventure) [reply libcontainerd/client_daemon.go]: Those requests would just fail until the connection monitor reconnects normally.

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Pull request merged**